### PR TITLE
Image installation during continuous warm reboot test

### PR DESCRIPTION
**Pull request by vaibhavhd**
### Description of PR

Summary: Image installation during continuous warm reboot test. Additionally, reorganizing the test to work in a Class-based design.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
The continuous warm reboot test presently executes the reboot cycles on image that is pre-installed on the DUT.
Reboot on the current image is not very helpful as the installed version soon gets obsolete. It would be much useful if new images can be installed at runtime (when the test is still executing).
#### How did you do it?
Image installation is now integrated with cont_warm_reboot_test. Two new CLI arguments introduced - ```image_location``` and ```image_list```.
The data is converted into JSON file which can be modified anytime when the test is still running.

#### How did you verify/test it?
Tested multiple 201811 image on T0 topology during single testcase execution. Started continuous warm reboot test with the current image. During the test, the JSON file was modified with new images to be installed. Test picked up new images and reboot cycles continued.

### CODE REVIEW

File: tests/common/fixtures/advanced_reboot.py [hunk 1]
```diff
@@ -382,6 +382,17 @@ def runRebootTestcase(self, prebootList=None, inbootList=None, prebootFiles=None
         # Handle mellanox platform
         self.__handleMellanoxDut()
 
+    def runRebootTestcase(self, prebootList=None, inbootList=None, prebootFiles=None):
+        '''
+        This method validates and prepare test bed for rebot test case. It runs the reboot test case using provided
```
Comment (SavchukRomanLv) [inline tests/common/fixtures/advanced_reboot.py:387]: Please fix tupo

File: tests/platform_tests/args/cont_warm_reboot_args.py [hunk 2]
```diff
@@ -25,3 +25,17 @@ def add_cont_warm_reboot_args(parser):
...
+        default=None,
+        help="Enable continuous IO",
+    )
+
+    parser.addoption(
+        "--image_list",
+        action="store",
+        default="current",
+        help="Comma separated list of images to be installed during continuous reboot test",
+    )
```
Comment (SavchukRomanLv) [inline tests/platform_tests/args/cont_warm_reboot_args.py:41]: Add newline

File: tests/platform_tests/test_cont_warm_reboot.py [hunk 3]
```diff
@@ -33,132 +34,168 @@
...
+
+        # Start advancedReboot script on the ptf host to enable continuous I/O
+        advancedReboot = self.get_advanced_reboot(rebootType='warm-reboot', enableContinuousIO=self.enable_continuous_io)
+        thr = threading.Thread(target=advancedReboot.runRebootTestcase)
+        thr.setDaemon(True)
+        thr.start()
+
+        file_template = {
+            'install_list': self.image_list, # this list can be modified at runtime to enable testing different images
+            'location': 'http://100.127.20.23/installer/sonic/broadcom/internal-201811/',
```
Comment (SavchukRomanLv) [inline tests/platform_tests/test_cont_warm_reboot.py:154]: Why this is hardcoded?

File: tests/platform_tests/test_cont_warm_reboot.py [hunk 4]
```diff
@@ -33,132 +34,168 @@
...
+        thr.join(60)
+        if thr.is_alive():
+            logging.error("Failed to join continuous I/O thread in 60s")
+        logging.info("Continuous warm-reboot test completed")
+
+def test_cont_warm_reboot(request, duthost, ptfhost, localhost, conn_graph_facts, get_advanced_reboot):
+    continuous_reboot = ContinuousReboot(request, duthost, ptfhost, localhost, conn_graph_facts, \
+        get_advanced_reboot)
+
+    continuous_reboot.start_cont_warm_reboot()
```
Comment (SavchukRomanLv) [inline tests/platform_tests/test_cont_warm_reboot.py:201]: Please, add newline

File: tests/common/fixtures/advanced_reboot.py (hunk 1 above)
Comment (vaibhavhd) [reply tests/common/fixtures/advanced_reboot.py]: Fixed, thanks.

File: tests/platform_tests/args/cont_warm_reboot_args.py (hunk 2 above)
Comment (vaibhavhd) [reply tests/platform_tests/args/cont_warm_reboot_args.py]: Fixed, thanks.

File: tests/platform_tests/test_cont_warm_reboot.py (hunk 3 above)
Comment (vaibhavhd) [reply tests/platform_tests/test_cont_warm_reboot.py]: This should not be hardcoded, the value of CLI argument `--image_location` is to be used here. Fixed now, thanks.

File: tests/platform_tests/test_cont_warm_reboot.py (hunk 4 above)
Comment (vaibhavhd) [reply tests/platform_tests/test_cont_warm_reboot.py]: Fixed, thanks.

File: tests/platform_tests/args/cont_warm_reboot_args.py [hunk 5]
```diff
@@ -25,3 +25,17 @@ def add_cont_warm_reboot_args(parser):
         default=False,
         help="Enable continuous IO",
     )
+
+    parser.addoption(
+        "--image_location",
+        action="store",
+        default=None,
+        help="Enable continuous IO",
```
Comment (tahmed-dev) [inline tests/platform_tests/args/cont_warm_reboot_args.py:33]: help phrase does not match option?

File: tests/common/fixtures/advanced_reboot.py [hunk 6]
```diff
@@ -382,6 +381,17 @@ def runRebootTestcase(self, prebootList=None, inbootList=None, prebootFiles=None
         # Handle mellanox platform
         self.__handleMellanoxDut()
```
Comment (tahmed-dev) [inline tests/common/fixtures/advanced_reboot.py:382]: why is the `imageInstall` a wrapper around `__HandleMellanoxDut`? is there a code missing here?

Comment (tahmed-dev) [reply tests/common/fixtures/advanced_reboot.py:382]: nevermind, I saw it being used below.

File: tests/platform_tests/args/cont_warm_reboot_args.py (hunk 5 above)
Comment (vaibhavhd) [reply tests/platform_tests/args/cont_warm_reboot_args.py]: Fixed now, thanks.

**Pull request merged**